### ISSUE TITLE
Replace SoapUI jar dependencies with our own implementation

### ISSUE THREAD

**Issue by lbroudoux** [microcks/microcks#884]
### Reason/Context

SoapUI jar and dependencies have been embedded into Microcks since day one as the support of Soap WebServices was (and is still) on the list of features. At that time, directly embedding SoapUI jars was a nice idea and was a way of getting faster to our goal. However, today, SoapUI brings a lot of challenges:

* Latest version `5.7.0` is 2 years old and cannot be integrated into Microcks because of conflicting libraries,
* Project is in maintenance mode and quite abandoned in favor or new priorities for SmartBear teams,
* The number of CVEs in dependency is huge,
* The number and volume of dependency - as well as their diversity licences to manage - is huge,
* The runtime behavior of SoapUI is not very pleasant - slowed by UI considerations that should not bother us in headless mode.

That's why we decide to start this initiative of replacing SoapUI implementations with our own. The goal is not to get rid of/discontinue support of SoapUI but rather to get even better control of it.

### Description

We are then considering replacing SoapUI jar and libraries with our own implementation as our use-cases are well-known and very well scoped. We are using SoapUI java APIs in 2 cases today:
* During the import of a SoapUI project to transform SoapUI model elements into Microcks one. The initial decision of using it to avoid Xml parsing "at hand" can be revised and this will just require a bunch of pure Xml parsing,
* During the test of an endpoint when running the `SOAPUI` strategy. This is actually using SoapUI classes to execute the different requests and attached assertions. This can also be easily replaced by our own `HttpTestRunner` base class but we have to port some SoapUI assertions into our own code-base.

For 1st use-case, we have a pretty descent unit tests coverage and it's just the matter of switching implementation while ensuring everything still running as is with the test suite.

The 2nd use-case is a bit trickier as we don't have an exhaustive view of the SoapUI test assertions that are actually used on the field... We propose starting with the implementation of the list below:
* Valid HTTP Status Codes
* SOAP Response
* SOAP Fault Assertion
* Simple Contains
* Not Contains
* Schema Compliance
* XPath Match
* JsonPath Match
* JsonPath Existence Match
* SLA

Please do not hesitate to comment and add new requirements regarding assertions!

cc @sfraigneau as I know you're a heavy user of this. The new implementation will stay in the repo next to the existing one in order to be able to easily switch them. I encourage you to test as early as possible and giving feedback!

**Comment by lbroudoux** [comment]
A new `quay.io/microcks/microcks:nightly` container image has been released without SoapUI and all its dependencies!
Here is the Quay.io summary before applying the commits:
![Capture d’écran 2023-08-03 à 13 21 50](https://github.com/microcks/microcks/assets/1538635/d5d312f3-821a-403a-9af8-f5eef12c9517)
And now with the commits applied:
![Capture d’écran 2023-08-03 à 15 18 33](https://github.com/microcks/microcks/assets/1538635/ca18f820-a361-4e13-aed2-3878997f4d65)

So basically:
* We decreased the size of the image by 30MB (that's close to 12%)
* We decreased the number of CVEs by 18 (that's close to 35%)
* The startup time of the container is now 2.0 sec instead of 2.5 sec on my machine (that's close to 20%)
* The memory consumption may also decrease as we're loading way fewer classes in Heap and Metaspace

**Comment by sfraigneau** [comment]
Hi Laurent,

> That's why we decide to start this initiative of replacing SoapUI implementations with our own. The goal is not to get rid of/discontinue support of SoapUI but rather to get even better control of it.

This is a very good idea. We are facing the same issues as you and I'm sure other people do. Can I suggest to create a dedicated library to handle the SoapUI stuff ?  It could be use in other projets than microcks.

I'll give it a try to the nighty build. Thanks for the update.

**Comment by lbroudoux** [comment]
Add some missing assertions this morning.

Good suggestion @sfraigneau but I'm afraid we'd still be sticky to microcks internal model as:
* import's goal is to build our domain objects,
* assertion mechanism tightly integrates with our representation of exchange (Request/Response pair) and resources (WSDL, XSD) attached to a service.

Maybe we could put everything related to soapui into a specific Maven module with minimal dependencies on `microcks-model` (plain Java beans) to ease maintenance and contributions? (and maybe allow external usage?) What do you hink?

**Comment by sfraigneau** [comment]
Hi Laurent,

I have tested the 1.18 branch and I did not see any issue to import our mocks. Congrats.
However, there is a problem in the mock script. If we use an xpath which refers to a namespace (i.e. //foo:bar[1]), it cannot finds anything. Not sure what the problem is. I saw in your implementation of the XmlHolder class that the namespaces in the xpath are turn on (which is good) but the declareNamespace method is never call. Is it something you planned to handle later ? If you need help to investigate, just tell me.
I'll do further tests next week and keep you inform.

Regarding the soapui lib, I did not think about these constraints. It would be difficult to create a generic one. 
By the way, your proposal is a good idea to isolate that part in a first step. We could discuss more on that later.

Have a good week-end

**Comment by lbroudoux** [comment]
Thanks for feedback @sfraigneau! It's super nice to see that things are close to the end. 

Regarding the issue you're facing: I didn't planned further additions to XmlHolder as I didn't detect any regression on my side... I would be definitively happy if you could help scope the problème and investigate with me!

Maybe we could plan a meeting next week to dig in this topic?

**Comment by sfraigneau** [comment]
Hi Laurent,

I fixed it in the MR and I did not see any other issues on my side :)

Yes, I'll contact you.

regards
